lib/pipeline-add-uuids.js

```diff
@@ -33,19 +33,21 @@ const connectionOptions = {
 const JSON_PARSER = new SparqlJsonParser();
 const UUID_CACHE = new LRUCache({
   max: 500000,
-  fetchMethod: async (key) => {
+  fetchMethod: async ({ key, selfCache }) => {
```

Comment:
I don't understand why this selfCache is needed, the LRU-cache should already cache values and fetchMethod should only be called if the key does not have a value in the cache... if it wasn't working before I think we might have an issue with our key.

Reply:
You're right. :facepalm: I implemented that `selfCache` thing before fixing the mirroring. The faulty mirroring was the actual root cause for double UUIDs as well. There is even a mistake with this implementation, so I will be removing it and updating the explanation on this PR.

Reply:
Although there is one more thing: this LRU cache is limited in size. If the amount of subjects per Task grows beyond the maximum size of this cache, then we might still see double UUIDs. If we encounter an already seen subject after seeing 500000 other subjects, it will still generate a new UUID for it.
With a size of 500000 this is unlikely.